### DISCUSSION TITLE
Financially support/sponsor?

### DISCUSSION THREAD

**Discussion by opsdisk** [mjcaley/aiospamc#316] [General]
Any way to financially support/sponsor this project through GitHub or another platform?

**Comment by opsdisk** [comment]
Tagging you @mjcaley in case you missed it.

**Comment by mjcaley** [comment]
Hi @opsdisk!

I honestly wasn't sure if anyone was seriously using the library so I hadn't considered it, but I would really appreciate the support.

I've been checking out the GitHub Sponsors program and it looks like it might be a good candidate.  I'll see if I can set something up.

Also, let me know if you have any comments. Some outside feedback would be great.

Thanks!

**Comment by opsdisk** [comment]
That works!  It's great library and worked better than https://github.com/petermat/spamassassin_client for my project.  Really appreciate the work you put into it.

I was running into some issues (it was a while ago so the details are hazy) since it was running in multiple threads (async isn't my strong suit, so it could have just been my misunderstanding of Python's async), but I had to tweak your example.  This is what I did and it seems to be running like a champ now.

```python
    # Since this is running in a thread, "loop = asyncio.get_event_loop()" won't work, need to create a new one.
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)

    responses = loop.run_until_complete(
        asyncio.gather(aiospamc.check(message, host="127.0.0.1", port=783, loop=loop))
    )

```

**Reply by mjcaley** [reply]
Ya, I think you ended up with the right solution.  As I understand it the asyncio event loop is bound to an OS thread.  So if you have several threads they'll each have to have their own event loop spun up.

**Comment by opsdisk** [comment]
Hi @mjcaley - saw you added sponsors!  Not sure if it's finalized, but any chance you could add a one-time custom payment option?

**Reply by mjcaley** [reply]
@opsdisk Sure I just enabled it.  I didn't see that the first time.

Thanks!

**Reply by opsdisk** [reply]
:smile: just sponsored you!  Thanks again for your work on this library!
